packages/common/src/common.ts

```diff
@@ -636,67 +702,45 @@ export class Common extends EventEmitter {
     return BigInt(ttd)
   }
 
-  /**
-   * True if block number provided is the hardfork (given or set) change block
-   * @param blockNumber Number of the block to check
-   * @param hardfork Hardfork name, optional if HF set
-   * @returns True if blockNumber is HF block
-   */
-  isHardforkBlock(blockNumber: BigIntLike, hardfork?: string | Hardfork): boolean {
-    blockNumber = toType(blockNumber, TypeOutput.BigInt)
-    hardfork = hardfork ?? this._hardfork
-    const block = this.hardforkBlock(hardfork)
-    return typeof block === 'bigint' && block !== BigInt(0) ? block === blockNumber : false
-  }
-
   /**
    * Returns the change block for the next hardfork after the hardfork provided or set
    * @param hardfork Hardfork name, optional if HF set
-   * @returns Block number or null if not available
+   * @returns Block timestamp, number or null if not available
```

Comment:
I'm a little anxious about the naming of this method now.  I know we don't really want a breaking release for `common` but its a complete misnomer for us to call `nextHardforkBlock` if we start returning timestamps as one possibility.  We should think very carefully about this and consider maybe adding a new function nameed something like `nextHardforkBlockOrTimestamp` (and then deprecate `nextHardForkBlock` in the same release with an intent to remove it in the next breaking release).  My biggest concern is that there's no way for the library consumer to know programmatically whether or not we're returning a block number or a timestamp.  I realize there are other factors in play whereby library consumers can know this (check the consensus type and see if PoS or other) but that wouldn't be reliable since it could vary across with network you're on.